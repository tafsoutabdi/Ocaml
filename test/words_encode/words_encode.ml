open Tokenization.Words.Tokenizer
open Utils.Assoc
open Utils.Std

let print_context voc s = 
  Format.printf "voc = %s@.s = \"%s\"@.encode voc s = "
    (string_of_vocabulary voc) s

let () = 
  let voc = [("hello", 13); (" ", 4); ("world", 7); ("!", 12)] in
  let s = "hello world!!" in
  print_context voc s;
  print_id_list (encode voc s);
  Format.printf "@."

let () = 
  let voc = [("hello", 13); ("world", 7); (" ", 4); ("!", 12)] in
  let s = "hello helloworld!!" in
  print_context voc s;
  try ignore @@ encode voc s
  with EncodingError s -> 
    Format.printf "EncodingError \"%s\"@." s 