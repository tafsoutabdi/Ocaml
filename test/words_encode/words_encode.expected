voc = [("hello", 13); (" ", 4); ("world", 7); ("!", 12)]
s = "hello world!!"
encode voc s = [13; 4; 7; 12; 12]
voc = [("hello", 13); ("world", 7); (" ", 4); ("!", 12)]
s = "hello helloworld!!"
encode voc s = EncodingError "helloworld!!"
